src/basic_memory/utils.py

```diff
@@ -27,7 +27,7 @@ def __str__(self) -> str: ...
 logging.getLogger("opentelemetry.sdk.metrics._internal.instrument").setLevel(logging.ERROR)
 
 
-def generate_permalink(file_path: Union[Path, str, PathLike]) -> str:
+def generate_permalink(file_path: Union[Path, str, PathLike], split_extension: bool = True) -> str:
```

Comment:
Most modifications in this file are simply lint fixes, applied automatically on save in my IDE.

The main changes are:
- adding the optional `split_extension` parameter
- retaining the file extension in a variable
- appending the extension back into the returned string, when applicable

Doing so allows this function to be reused, keeping things DRY, rather than repeating almost the entire function for use when generating titles.